sourcegraph/sourcegraph
# (feat) add repo meta write rbac

Part of https://github.com/sourcegraph/pr-faqs/issues/96.

This PR:
- adds `REPO_METADATA#WRITE` RBAC
- adds RBAC check for repo metadata create/delete/update 
- adds RBAC check for UI repo metadata edit page

## Test plan
- `sg start`
- Enable `repository-metadata` feature flag
- Go to any repository page and check that there is a gear icon/link available near metadata. Click it and it should open repo metadata edit page
- Uncheck `REPO_METADATA#WRITE` permission for your current non-admin user https://sourcegraph.test:3443/site-admin/roles. NOTE: that admin users will have all permissions and they are not revokable
- Go to any repository page and check that there is NO gear icon/link available near metadata

## Screenshots

https://github.com/sourcegraph/sourcegraph/assets/6717049/73439b39-2d40-4dbd-88a5-c4d43be504eb
